### ISSUE TITLE
Forbid using multiple `for` in comprehensions

### ISSUE THREAD

**Issue by sobolevn** [wemake-services/wemake-python-styleguide#255]
# Rule request

## Thesis

You can write very complex comprehensions like this:

```python
upper_case_violations = [
        target
        for assignment in top_level_assigns
        for target in assignment.targets
        for _ in range(10)
        if isinstance(target, ast.Name) and is_upper_case_name(target.id)
    ]
```

And it will be allowed by all complexity checks.

## Reasoning

This is a perfect example of how you should not do.

1. This expression contains three nested loops, which indicates something like `O(n**3)`
2. It is unreadable
3. It can be refactored be better

We should catch comprehensions with more than two `for` loops. And report them as too complex.

**Comment by adityashrm21** [comment]
How to report these for loop comprehensions?

**Comment by sobolevn** [comment]
Closed via https://github.com/wemake-services/wemake-python-styleguide/pull/270#pullrequestreview-165589901
